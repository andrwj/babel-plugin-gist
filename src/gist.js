// @flow
import client from 'cheerio-httpcli'

export default (id: string) => {
  const files = id.split('#')
  const url = files[0]
  // const user = url[0]
  // const hash = url[1]
  const name = files[1]
  const selector = name ? `#${name} .highlight` : '.highlight'

  return client
    .fetchSync(`https://gist.github.com/${url}`)
    .$(selector)
    .text()
}
