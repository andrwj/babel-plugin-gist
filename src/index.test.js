// @flow
import path from 'path'
import rimraf from 'rimraf'
import pluginTester from 'babel-plugin-tester'
import plugin from '.'

jest.mock('./gist')
import gist from './gist' // eslint-disable-line

const code = `
export default (a, b) => a + b
`
// $FlowFixMe
gist.mockImplementation(() => code)

afterAll(() => {
  rimraf.sync(path.resolve(__dirname, '..', 'node_modules/.cache/'))
})

pluginTester({
  title: 'default',
  plugin,
  snapshot: true,
  tests: [
    `import sum from 'gist:rhiokim/badfbe21fb86dbbe031cac85bdc2043a#file-sum-js'`,
    {
      title: 'use cache',
      code: `
    import sum from 'gist:rhiokim/badfbe21fb86dbbe031cac85bdc2043a#file-sum-js'
    `,
    },
    {
      title: 'typo',
      code: `
    import sum from 'giiist:rhiokim/badfbe21fb86dbbe031cac85bdc2043a#file-sum-js'
    `,
    },
  ],
})
