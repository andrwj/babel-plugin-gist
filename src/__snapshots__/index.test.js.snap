// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1. gist 1`] = `
"
import sum from 'gist:rhiokim/badfbe21fb86dbbe031cac85bdc2043a#file-sum-js'

      ↓ ↓ ↓ ↓ ↓ ↓

const sum = (a, b) => a + b;
"
`;

exports[`typo 1`] = `
"
import sum from 'giiist:rhiokim/badfbe21fb86dbbe031cac85bdc2043a#file-sum-js'

      ↓ ↓ ↓ ↓ ↓ ↓

import sum from 'giiist:rhiokim/badfbe21fb86dbbe031cac85bdc2043a#file-sum-js';
"
`;

exports[`use cache 1`] = `
"
import sum from 'gist:rhiokim/badfbe21fb86dbbe031cac85bdc2043a#file-sum-js'

      ↓ ↓ ↓ ↓ ↓ ↓

const sum = (a, b) => a + b;
"
`;
